### PR TITLE
PLAT-102253: Migrate spotlight/Spottable to use hooks

### PR DESCRIPTION
**Pull request by ybsung**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

Migrate spotlight/Spottable to use hooks

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

Add `Spot` for cancelable behavior.
Add `useSpot` hook to manage the life cycle of the Spottable behavior.
Adopt `spotlight/Spottable` to use `useSpot`.
Add `useSpot-specs.js`.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

1. I updated the default value to `true` for the `emulateMouse` in the JS Doc.
2. I renamed `focusedWhenDisabled` as `isFocusedWhenDisabled` like `isFocused`.
3. <Question> Do we need to set `selectCancelled` to `false` when `Spot.unload()` is called?
4. `onMouseUp` was forwarded in the `componentDidUpdate` before. To do it with the `useSpot`, the `onSelectionCancel` prop with `onMouseUp` function is passed to the `useSpot` now.

Before
```
// Spottable.js
componentDidUpdate (prevProps) {
	...
	if (this.isFocused && this.props.disabled && lastSelectTarget === this && !selectCancelled) {
		selectCancelled = true;
		forward('onMouseUp', null, this.props);
	}
```
After
```
// Spottable.js
const spot = useSpot({..., onSelectionCancel: rest.onMouseUp, ...});

// Spot.js
didUpdate = () => {
	...
	if (this.isFocused && this.props.disabled && lastSelectTarget === this && !selectCancelled) {
		selectCancelled = true;
		forward('onSelectionCancel', null, this.props);
	}
```
5. There is the concern about my comment ( https://github.com/enactjs/enact/pull/2758#discussion_r412094605 ) for the following Warning.
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
      
      Check the render method of `Spottable`.
          in mockConstructor (at Spottable.js:148)
          in Spottable (created by WrapperComponent)
          in WrapperComponent
```
### Links
[//]: # (Related issues, references)

PLAT-102253

### CODE REVIEW

File: packages/spotlight/Spottable/useSpot.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	const mutableRef = React.useRef({
+		prevSpotlightDisabled: spotlightDisabled,
+		spotlightDisabled
+	});
+
+	spot.setContext(mutableRef);
+
+	React.useEffect(() => {
+		// eslint-disable-next-line react/no-find-dom-node
+		spot.load(ReactDOM.findDOMNode(ref.current));
```
Comment (ybsung) [inline packages/spotlight/Spottable/useSpot.js:70]: Hello @ryanjduffy 

I had one problem to migrate Spottable to use Hooks.
In detail, I don't know how to change the `ReactDOM.findDOMNode(this)` properly on the `useSpot`. With the PR, there is the following warning. Do we have to keep using the `Spottable` as the React Component?

```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
      
      Check the render method of `Spottable`.
          in mockConstructor (at Spottable.js:148)
          in Spottable (created by WrapperComponent)
          in WrapperComponent
```

Comment (ybsung) [reply packages/spotlight/Spottable/useSpot.js:70]: In my opinion, there is no problem that the `Spottable` works. And there are no errors to run lint in enact and sandstone. So could we go ahead and fix it later? Or should we have to fix it in this PR?

File: packages/spotlight/Spottable/Spottable.js [hunk 2]
```diff
@@ -65,7 +38,7 @@ const defaultConfig = {
 	 * to Spotlight 5-way events.
 	 *
 	 * @type {Boolean}
-	 * @default false
```
Comment (webOS101) [inline packages/spotlight/Spottable/Spottable.js:68]: :doh:

File: packages/spotlight/Spottable/useSpot.js (hunk 1 above)
Comment (ybsung) [reply packages/spotlight/Spottable/useSpot.js]: To maintain `ref` properly, I defined `Spottable` based on React.Component and passed the `componentRef` from `ReactDOM.findDOMNode(this)` to `SpottableBase` and `useSpot`.

```
	class Spottable extends React.Component {
		componentDidMount () {
			// eslint-disable-next-line react/no-find-dom-node
			this.node = ReactDOM.findDOMNode(this);
		}

		get componentRef () {
			return this.node;
		}

		render () {
			return <SpottableBase {...this.props} componentRef={this.componentRef} />;
		}
	}
```

**Review by 0x64 — APPROVED** [review]
:lgtm:

File: packages/spotlight/Spottable/Spottable.js [hunk 3]
```diff
@@ -96,353 +103,178 @@ const defaultConfig = {
...
+			onSpotlightDisappear,
+			onSpotlightDown,
+			onSpotlightLeft,
+			onSpotlightRight,
+			onSpotlightUp,
+			selectionKeys,
+			spotlightDisabled,
+			...rest
+		} = props;
+		const spot = useSpot({
```
Comment (jeonghee27) [inline packages/spotlight/Spottable/Spottable.js:120]:
```suggestion
		const hook = useSpot({
```

**Pull request merged**